tuning/autotune.py

```diff
@@ -74,15 +157,42 @@ class TaskResult:
 
 
 @dataclass
-class BenchmarkOutput:
-    output_str: Optional[str] = None
+class DispatchBenchmarkResult:
+    result_str: str = None
+
+    @property
```

Comment:
For this function and the ones below, I wouldn't make them properties if the computation can fail. I only use properties for functions with very very simple logic that always succeed.